web/src/settings_users.ts

```diff
@@ -191,29 +191,49 @@ function role_selected_handler(
     widget.render();
 }
 
-function get_roles(): {
+function count_users_by_role(user_ids: number[]): Record<number, number> {
+    const role_counts: Record<number, number> = {};
+
+    for (const user_id of user_ids) {
+        const user = people.get_by_user_id(user_id);
+        const role_code = user.role;
+
+        role_counts[role_code] = (role_counts[role_code] ?? 0) + 1;
+    }
+
+    return role_counts;
+}
+
+type RoleOption = {
```

Comment:
This type is used only at one place, so we can probably avoid defining this now.